packages/palette/src/elements/ModalDialog/ModalDialogDialog.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react"
+import { Clickable } from "../Clickable"
+import { DROP_SHADOW } from "../../helpers/shadow"
+import { Box, BoxProps } from "../Box"
+import { Flex } from "../Flex"
+import { Text } from "../Text"
+import { CloseIcon } from "../../svgs/CloseIcon"
+import { ArtsyLogoBlackIcon } from "../../svgs/ArtsyLogoBlackIcon"
+import { useSentinelVisibility } from "../../utils/useSentinelVisibility"
+import { Spacer } from "../Spacer"
+
+export interface ModalDialogDialogProps extends BoxProps {
+  children: React.ReactNode
+  footer?: React.ReactNode
+  hasLogo?: boolean
+  onClose: () => void
+  title?: string
+}
+
+export const ModalDialogDialog: React.FC<ModalDialogDialogProps> = ({
+  children,
+  footer,
+  hasLogo,
+  onClose,
+  title,
+  ...rest
+}) => {
+  const {
+    sentinel: topSentinel,
+    isSentinelVisible: isAtTop,
+  } = useSentinelVisibility()
+
+  const {
+    sentinel: bottomSentinel,
+    isSentinelVisible: isAtBottom,
+  } = useSentinelVisibility()
+
+  return (
+    <Flex
+      bg="white100"
+      flexDirection="column"
+      overflow="hidden"
+      m={2}
+      style={{ boxShadow: DROP_SHADOW }}
+      {...rest}
+    >
+      <Flex
+        alignItems="flex-start"
+        justifyContent="space-between"
+        style={{
+          transition: "box-shadow 250ms",
+          boxShadow: isAtTop ? DROP_SHADOW : undefined,
+        }}
+      >
+        {(title || hasLogo) && (
+          <Box m={2}>
+            {hasLogo && (
+              <ArtsyLogoBlackIcon
+                display="block"
+                style={{ width: "75px", height: "26px" }}
```

Comment:
This manually sizes the icon using with the exact aspect ratio to avoid excessive margin.